R/vector_control_parameters.R

```diff
@@ -133,3 +133,52 @@ set_spraying <- function(
   parameters$spraying_ms_gamma <- ms_gamma
   parameters
 }
+
+#' @title Parameterise a habitat management strategy
+#'
+#' @description The model will simulate larval source habitat management at 
+#' `timesteps` to the entire human population. 
+#' The impact will reduce adult mosquitoes recruited from pupae to the susceptible
+#' cohort.
+#'
+#' The rapidity of the impact is captured by the lsm_factor function
+#' by changing the parameters lsm_rate and deprec_param
+#'
+#' The amount of impact in terms of reduced adult mosquito densities
+#' that the habitat management has is determined by larvi_min which is
+#' set to 1 as default when there is no impact.
+#'
+#' The structure for the habitat model is documented in the
+#' working draft Sherrard-Smith et al (larviciding in Kenya)
+#'
+#' @param parameters a list of parameters to modify
+#' @param timesteps the timesteps at which to distribute lsm rounds
+#' @param larvi_min the proportion of the mosquito population reduced (0 to 1), 1 is no impact
+#' @param larvi_min a matrix of reduction probabilities for each species over time
+#' With nrows=length(timesteps), ncols=length(species)
+#' @param lsm_rate a matrix of reduction probabilities for each species over time
+#' With nrows=length(timesteps), ncols=length(species)
+#' @param deprec_param a matrix of reduction probabilities for each species over time
+#' @export
+set_habitat_management <- function(
```

Comment:
I wonder if there should be a check within this function that `parameters$individual_mosquitoes = FALSE`?
Currently I can set all this stuff and have it TRUE and it appears to all be fine.

Reply:
this is a great idea - could you help to code that?